tests/integration_test/shadow_system_test.c

```diff
@@ -0,0 +1,932 @@
+                                                          &socketsConfig );
+
+        if( transportStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+        {
+            LogWarn( ( "Connection to the broker failed. Retrying connection with backoff and jitter." ) );
+            retryUtilsStatus = RetryUtils_BackoffAndSleep( &reconnectParams );
+        }
+        else
+        {
+            isSuccessful = true;
+        }
+
+        if( retryUtilsStatus == RetryUtilsRetriesExhausted )
+        {
+            LogError( ( "Connection to the broker failed, all attempts exhausted." ) );
+        }
+    } while( ( transportStatus != TRANSPORT_SOCKET_STATUS_SUCCESS ) && ( retryUtilsStatus == RetryUtilsSuccess ) );
+
+    return isSuccessful;
+}
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/**
+ * @brief Test group for device Shadow system tests.
+ */
+TEST_GROUP( deviceShadow_Integration );
+
+TEST_SETUP( deviceShadow_Integration )
+{
+    /* Reset file-scoped global variables. */
+    receivedSubAck = false;
+    receivedUnsubAck = false;
+    receivedPubAck = false;
+    receivedPubRec = false;
+    receivedPubRel = false;
+    receivedPubComp = false;
+    persistentSession = false;
+    receivedUpdateDeltaResult = false;
+    receivedUpdateDocumentsResult = false;
+    receivedUpdateAcceptedResult = false;
+    receivedUpdateRejectedResult = false;
+    receivedDeleteAcceptedResult = false;
+    receivedDeleteRejectedResult = false;
+    receivedGetAcceptedResult = false;
+    receivedGetRejectedResult = false;
+
+    /* Establish TLS over TCP connection with retry attempts on failures. */
+    TEST_ASSERT_TRUE( connectToServerWithBackoffRetries( &networkContext ) );
+
+    /* Establish MQTT session on top of the TCP+TLS connection. */
+    establishMqttSession( &context, &networkContext, true, &persistentSession );
+}
+
+TEST_TEAR_DOWN( deviceShadow_Integration )
+{
+    /* Terminate MQTT connection. */
+    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+
+    /* Terminate TLS session and TCP connection. */
+    ( void ) SecureSocketsTransport_Disconnect( &networkContext );
+}
+
+/* ========================== Test Cases ============================ */
+
+
+
+/**
+ * @brief Test group runner for MQTT system tests that can be run against AWS IoT.
+ */
+TEST_GROUP_RUNNER( deviceShadow_Integration )
+{
+    RUN_TEST_CASE( deviceShadow_Integration, test_Shadow_System );
+}
+
+/**
+ * @brief Subscribes the shadow topics: /update/delta, /update/documents,
+ * /update/accepted, /delete/accepted, /get/accepted, then publish the
+ * regarding payloads to verify if receiving the notification from the
+ * subscribed topics.
+ */
+TEST( deviceShadow_Integration, test_Shadow_System )
+{
+    /* A buffer containing the update document. It has static duration to prevent
+     * it from being placed on the call stack. */
+    static char updateDocument[ 1 ] = { 0 };
```

Comment:
A better name would be `nullDocument` or `emptyDocument`

Reply:
thanks for the suggestion. I'll take emptyDocument.